mlflow/pyfunc/__init__.py

```diff
@@ -2912,29 +2912,39 @@ def predict(self, context, model_input: List[str], params=None) -> List[str]:
                 CHAT_MODEL_INPUT_SCHEMA,
                 CHAT_MODEL_OUTPUT_SCHEMA,
             )
-            input_example = input_example or CHAT_MODEL_INPUT_EXAMPLE
-            input_example, input_params = _split_input_data_and_params(input_example)
-
-            if isinstance(input_example, list):
-                params = ChatParams()
-                messages = []
-                for each_message in input_example:
-                    if isinstance(each_message, ChatMessage):
-                        messages.append(each_message)
-                    else:
-                        messages.append(ChatMessage.from_dict(each_message))
+            if input_example:
+                input_example, input_params = _split_input_data_and_params(input_example)
+                valid_params = {}
+                if isinstance(input_example, list):
+                    messages = []
+                    for each_message in input_example:
+                        if isinstance(each_message, ChatMessage):
+                            messages.append(each_message)
+                        else:
+                            messages.append(ChatMessage.from_dict(each_message))
+                else:
+                    # If the input example is a dictionary, convert it to ChatMessage format
+                    messages = [
+                        ChatMessage.from_dict(m) if isinstance(m, dict) else m
+                        for m in input_example["messages"]
+                    ]
+                    valid_params = {
+                        k: v
+                        for k, v in input_example.items()
+                        if k != "messages" and k in ChatParams.keys()
+                    }
```

Comment:
should we warn if user provides invalid params in their input example?

Reply:
i guess we don't warn if there are invalid keys in the ChatMessage, maybe this can be some follow-up work

Reply:
Yes we will warn at pyfunc prediction stage if it's not in the signature though.
There's a tracked item to validate input example against signature (if provided), and it would work for all models.